plugins/org.eclipse.n4js.utils/src/org/eclipse/n4js/utils/URIUtils.java

```diff
@@ -272,4 +263,35 @@ public static URI addEmptyAuthority(URI uri) {
 		return uri;
 	}
 
+	/** Converts the given {@link org.eclipse.emf.common.util.URI} to a {@link java.net.URI} */
+	static public java.net.URI toURI(URI uri) throws URISyntaxException {
+		return new java.net.URI(uri.toString());
+	}
+
+	/**
+	 * Converts the given URI to a {@link File} iff it is a {@link org.eclipse.emf.common.util.URI#isFile() file URI};
+	 * otherwise returns <code>null</code>.
+	 * <p>
+	 * Same as {@link org.eclipse.emf.common.util.URI#toFileString()}, but returns a {@link File} instance instead of a
+	 * string.
+	 */
+	static public File toFile(URI uri) {
+		return uri != null && uri.isFile() ? new File(uri.toFileString()) : null;
+	}
+
+	/**
+	 * Converts the given URI to a {@link Path} iff it is a {@link org.eclipse.emf.common.util.URI#isFile() file URI};
+	 * otherwise returns <code>null</code>.
+	 * <p>
+	 * Same as {@link org.eclipse.emf.common.util.URI#toFileString()}, but returns a {@link Path} instance instead of a
+	 * string.
+	 * <p>
+	 * <b>Attention: This fails on platform Windows:</b><br>
+	 * {@code Paths.get(org.eclipse.emf.common.util.URI#toFileString());} <br>
+	 * The reason is that {@link URI#toFileString()} returns a string like "\\c:\\dir" which cannot be parsed by
+	 * {@link Paths#get(java.net.URI)} due to the starting "\\".
+	 */
+	static public Path toPath(URI uri) {
+		return toFile(uri).toPath();
```

Comment:
Please check `SafeURI` and its sub-classes (esp. `FileURI`) if there is code that duplicates the code in the above utility methods. In that case, it probably makes sense to delegate to this class.

For example, I found:
https://github.com/eclipse/n4js/blob/b12c83bd1c2c1902087f7ed4fc6c80df4d979546/plugins/org.eclipse.n4js/src/org/eclipse/n4js/projectModel/locations/FileURI.java#L87